Extensions/blacklist.js

```diff
@@ -680,64 +689,98 @@ XKit.extensions.blacklist = new Object({
+		}
 
-		var to_add_type = "";
+		if (XKit.extensions.blacklist.preferences.mini_block.value !== true) {
+			$(obj).addClass("xblacklist_blacklisted_post_full_ui");
+		}
 
-		if (XKit.extensions.blacklist.preferences.show_type.value === true) {
+		if (XKit.extensions.blacklist.preferences.show_tags.value === true && XKit.extensions.blacklist.preferences.mini_block.value === false) {
+			$(obj).find(".post_footer").css('display', 'none');
+		} else {
+			$(obj).find(".post_tags, .post_footer, .mh_post_tags").css('display', 'none');
+		}
 
-			to_add_type = "<div class=\"xkit-blacklist-post-type  " + $(obj).attr('data-type') + "\">&nbsp;</div>";
+		if ($(obj).hasClass("xkit-shorten-posts-shortened") === true) {
+
+			// This was also shortened.
+			$(obj).attr('data-xkit-blacklist-old-height', $(obj).css("height"));
+			$(obj).css("height", "auto");
+			$(obj).find(".xkit-shorten-posts-embiggen").css("display", "none");
 
 		}
+	},
 
+	hide_desktop_content: function(obj, word) {
+		var old_content = '<div style="display: none;" class="xblacklist_old_content">' +
+					$(obj).find(".post_content").html() + '</div>';
+    
 		var post_id = $(obj).attr('id');
 		// If the post doesn't have an id, reuse a conveniently unique attribute
 		if (!post_id) {
 			post_id = $(obj).attr('data-post-id');
 			$(obj).attr('id', post_id);
 		}
+    
+		var block_excuse = XKit.extensions.blacklist.get_block_excuse(word, $(obj).attr('data-type'), post_id);
 
-		var block_excuse = '<div class="xblacklist_excuse">' +
-					'Blocked because it contains the word "<b>' + word + '</b>"'  + to_add_type +
-					'<div data-post-id="' + post_id + '" class="xblacklist_open_post xkit-button">Show it anyway</div></div>';
-
-		if (XKit.extensions.blacklist.preferences.dont_show_cause.value === true) {
-			block_excuse = '<div class="xblacklist_excuse">' +
-					'Post blocked.' + to_add_type +
-					'<div data-post-id="' + post_id + '" class="xblacklist_open_post xkit-button">Show it anyway</div></div>';
-		}
-
-		$(obj).addClass("xblacklist_blacklisted_post");
 		$(obj).find(".post_info").css("display", "none");
 		$(obj).find(".post_controls").css("display", "none");
 		$(obj).find(".post_content").html(old_content + block_excuse);
 		$(obj).find(".post_footer_links").css('display', 'none');
 		$(obj).find(".post_source").css('display', 'none');
 		$(obj).find(".post-source-footer").css('display', 'none');
+		$(obj).find(".post_answer").css("display", "none");
+	},
 
-		if (XKit.extensions.blacklist.preferences.mini_block.value !== true) {
-			$(obj).addClass("xblacklist_blacklisted_post_full_ui");
-		}
-
-		if (XKit.extensions.blacklist.preferences.show_tags.value === true && XKit.extensions.blacklist.preferences.mini_block.value === false) {
-			$(obj).find(".post_footer").css('display', 'none');
-		} else {
-			$(obj).find(".post_tags, .post_footer").css('display', 'none');
+	hide_mobile_content: function(obj, word) {
```

Comment:
Is there a reason that this is in a function instead of using the same comma selector method the earlier stuff does? It seems like the logic is very similar here.